riscv/interactive.cc

```diff
@@ -467,38 +487,39 @@ void sim_t::interactive_str(const std::string& cmd, const std::vector<std::strin
 
   char ch;
   while((ch = mmu->load_uint8(addr++)))
-    putchar(ch);
+    sout << ch;
 
-  putchar('\n');
+  sout << endl;
 }
 
-void sim_t::interactive_until_silent(const std::string& cmd, const std::vector<std::string>& args)
+void sim_t::interactive_until_silent(const string& cmd, const std::vector<string>& args)
 {
   interactive_until(cmd, args, false);
 }
 
-void sim_t::interactive_until_noisy(const std::string& cmd, const std::vector<std::string>& args)
+void sim_t::interactive_until_noisy(const string& cmd, const std::vector<string>& args)
 {
   interactive_until(cmd, args, true);
 }
 
-void sim_t::interactive_until(const std::string& cmd, const std::vector<std::string>& args, bool noisy)
+void sim_t::interactive_until(const string& cmd, const std::vector<string>& args, bool noisy)
 {
   bool cmd_until = cmd == "until" || cmd == "untiln";
 
-  if(args.size() < 3)
+  if (args.size() < 3)
     return;
 
-  reg_t val = strtol(args[args.size()-1].c_str(),NULL,16);
-  if(val == LONG_MAX)
-    val = strtoul(args[args.size()-1].c_str(),NULL,16);
+  char *end;
```

Comment:
I don't understand this change with `end`; was something broken here before? Could this be a separate commit, where the commit message could explain this change?

Reply:
replaced `char *end` by NULL

Reply:
fixed in original commit